src/map/pc.hpp

```diff
@@ -676,7 +682,7 @@ struct map_session_data {
 	std::vector<int> cloaked_npc;
 
 	/* ShowEvent Data Cache flags from map */
-	bool *qi_display;
+	struct s_qi_display *qi_display;
```

Comment:
```suggestion
	std::vector<std::shared_ptr<s_qi_display>> qi_display;
```

This step will require a bit of logic cleanup `pc_show_questinfo_reinit`.